api/auth/security.py

```diff
@@ -3,12 +3,13 @@
 from flask import request, abort
 from flask_api import status
 from api import settings
-from api.auth.cas_auth import validate_proxy, validate_bearer, validate_cas_request
+from api.auth.cas_auth import validate_proxy, validate_bearer, validate_cas_request, validate_third_party
 from api.maap_database import db
 from api.models.member import Member
 from api.models.role import Role
 
 HEADER_PROXY_TICKET = "proxy-ticket"
+THIRD_PARTY_AUTH_HEADER = "X-Gitlab-Token"
```

Comment:
Is the assumption here that there will only be 1 third party? Maybe we could change it to be more explicit something like
```suggestion
THIRD_PARTY_AUTH_HEADER_GITLAB = "X-Gitlab-Token"
```